test/fluree/db/branch_test.clj

```diff
@@ -0,0 +1,280 @@
+            [fluree.db.test-utils :as test-utils]
+            [fluree.db.util :as util]))
+
+(deftest ^:integration branch-basic-operations-test
+  (testing "Basic branch operations with memory connection"
+    (let [conn @(fluree/connect-memory {})]
+      (try
+        ;; Create ledger and add initial data
+        (testing "Create ledger with initial data"
+          @(fluree/create conn "test-ledger" {})
+          @(fluree/insert! conn "test-ledger"
+                           {"@context" {"ex" "http://example.org/"}
+                            "@graph" [{"@id" "ex:alice"
+                                       "@type" "Person"
+                                       "ex:name" "Alice"}]}))
+
+        ;; Query main branch
+        (testing "Verify initial state in main branch"
+          (let [main-db @(fluree/load conn "test-ledger")
+                results @(fluree/query main-db {"@context" {"ex" "http://example.org/"}
+                                                "select" "?name"
+                                                "where" {"@id" "ex:alice" "ex:name" "?name"}})]
+            (is (= ["Alice"] results) "Should find Alice in main branch")))
+
+        (testing "Create feature branch from main"
+          (let [branch-result @(fluree/create-branch! conn "test-ledger:feature-branch" "test-ledger:main")]
+            (is (= "feature-branch" (:name branch-result)))
+            (is (= "main" (get-in branch-result [:created-from :branch])))
+            (is (some? (:created-at branch-result)))
+            (is (some? (:head branch-result))))
+
+          (testing "Add data to feature branch"
+            @(fluree/insert! conn "test-ledger:feature-branch"
+                             {"@context" {"ex" "http://example.org/"}
+                              "@graph" [{"@id" "ex:bob"
+                                         "@type" "Person"
+                                         "ex:name" "Bob"}]})
+
+            (testing "Verify branch isolation"
+              ;; Main branch should only have Alice
+              (let [main-db @(fluree/load conn "test-ledger:main")
+                    main-results @(fluree/query main-db {"@context" {"ex" "http://example.org/"}
+                                                         "select" "?name"
+                                                         "where" {"@id" "?id" "ex:name" "?name"}})]
+                (is (= ["Alice"] main-results) "Main branch should only have Alice"))
+
+              ;; Feature branch should have both Alice and Bob
+              (let [feature-db @(fluree/load conn "test-ledger:feature-branch")
+                    feature-results @(fluree/query feature-db {"@context" {"ex" "http://example.org/"}
+                                                               "select" "?name"
+                                                               "where" {"@id" "?id" "ex:name" "?name"}})]
+                (is (= (set feature-results) #{"Alice" "Bob"})
```

Comment:
This one has the expect/actual args backwards, and the results should be deterministic so no need for the `set` call.